js/this-inside-foo-bar/post.md

```diff
@@ -0,0 +1,81 @@
+# Title - What is `this` inside foo.bar()? 
+
+## The problem
+
+![Code showing - method which is passed as callback loses reference to original obj as this, but calling obj.method() directly works](https://cdn.hashnode.com/res/hashnode/image/upload/v1584584613992/rwNrFCElj.png)
+
+When we pass a method as callback, it loses reference to the original object (as `this`) when it is called.
+But calling a method directly works as expected.
+
+Why is that? **Can the spec help us explain this difference?**
+
+*(For simplicity, let's assume the caller doesn't intend to pass a different `this` value here, and calls onClick() directly)*
+
+------
+
+## The explanation
+
+##### For methods,
+⭐️ `foo.​bar()` translates to `foo.bar.call(foo)`.
+
+That is, when a method (function which is accessed as a object property) is called, the object gets passed as `this` inside the function.
+
+So, in this case -  
+*What* →  foo is passed as `this` inside bar  
+*When* →  if bar is a function and *accessed as object property* (i.e. foo.bar).  
+
+##### For functions, 
```

Comment:
adding example would be better. You can also add that when function is called in global context, `this` refers to `window`

Reply:
this value mentioned here was wrong. Fixed that.